src/pages/shop/Shop.tsx

```diff
@@ -1,15 +1,193 @@
+                      </Stack>
+                  </div>
+                  <div className="productColor">
+                    <p>Product Color</p>
+                    <div className="color-groups">
+                      <button className="color colorOne"></button>
+                      <button className="color colorTwo"></button>
+                      <button className="color colorThere"></button>
+                      <button className="color colorFour"></button>
+                      <button className="color colorFive"></button>
+                      <button className="color colorSix"></button>
+                      <button className="color colorSeven"></button>
+                      <button className="color colorEight"></button>
+                      <button className="color colorNine"></button>
+                      <button className="color colorTen"></button>
+                      <button className="color colorEleven"></button>
+                      <button className="color colorTwelve"></button>
+                      <button className="color colorThereteen lastest"></button>
+                    </div>
+                  </div>
+                  <div className="productSize">
+                    <p>Product Size</p>
+                    <div className="btnGroups">
+                      <button className='large'>Large</button>
+                      <button className='medium'>Medium</button>
+                      <button className='small'>Small</button>
+
+                    </div>
+                  
+                  </div>
+                  <div className="filterBySize">
+                    <p>Filter By Size</p>
+                  
+                  </div>
+                </Flex>
+              </div>
+            </button>
+
+            <div className="results">
+              <p>Showing 1–12 of 27 results</p>
+            </div>
+            <div className="card-view">
+              <button>
+                <BsFillGridFill className="grid gridOne" />
+              </button>
+              <button>
+                <BsFillGrid3X3GapFill className="grid gridTwo" />
+              </button>
+              <button>
+                <TfiLayoutGrid4Alt className="grid gridThere" />
+              </button>
+            </div>
+            <div className="sort">
+              <button className="default-sorting" onClick={handleToggle}>
+                <p>Default sorting</p>
+                <ChevronRight className="rightIco" />
+                <div id="sortingList" className={isActive ? 'active' : ''}>
+                  <div className="item">
+                    <p>Default sorting</p>
+                  </div>
+                  <p>Sort by popularity</p>
+                  <p>Sort by avarage rating</p>
+                  <p>Sort by lastest</p>
+                  <p>Sort by price:low to high</p>
+                </div>
+              </button>
+            </div>
+          </Flex>
+        </div>
+      </div>
```

Comment:
in general, you must divide this component into smaller parts, 190 lines are really too many.
by the way, each part might be a standalone task, so it might be done in paralel